architecture/2024-08-dashboard-component-threat-model.md

```diff
@@ -0,0 +1,299 @@
+```
+
+2. not expose backend directly.
+
+```
+app-config.yaml
+
+backend:
+  # Note that the baseUrl should be the URL that the browser and other clients
+  # should use when communicating with the backend, i.e. it needs to be
+  # reachable not just from within the backend host, but from all of your
+  # callers. When its value is "http://localhost:7007", it's strictly private
+  # and can't be reached by others.
+  baseUrl: http://localhost:7007
+  # The listener can also be expressed as a single <host>:<port> string. In this case we bind to
+  # all interfaces, the most permissive setting. The right value depends on your specific deployment.
+  listen: ':7007'
+```
+
+1.  use sqlite DB. This is light weight, not accessible to users directly and contains no useful information.
+  
+```
+app-config.yaml :
+
+backend:
+  database:
+    client: better-sqlite3
+    connection: ':memory:'
+```
+
+At present, Dashboard can only present the Radius application metadata visually. It has no ability to Create, Modify, Update or Delete any of the Radius application resources. This eliminates the scope of threats that require "write" action using the Dashboard portal.
+
+#### Storage of secrets
+
+None
+
+#### Data Serialization / Formats
+
+None
+
+### Cryptography 
+
+None
+
+### Clients
+
+The primary user of Dashboard is browser. At present, we dont have any other Backstage plugin that cloud be a Radius Dashboard client but that could change in future. 
+
+## Trust Boundaries
+
+We have a few different trust boundaries for the Dashboard component:
+
+- **Kubernetes Cluster**: The overall environment where the Dashboard component operates and serves clients.
+- **Namespaces within the Cluster**: Logical partitions within the cluster to separate and isolate resources and workloads.
+
+#### Key Points of Namespaces
+
+1. **Isolation of Resources and Workloads**: Different namespaces separate and isolate resources and workloads within the Kubernetes cluster.
+2. **Access Controls and Permissions**: Access controls and other permissions are implemented to manage interactions between namespaces.
+3. **Separation of Concerns**: Namespaces support the separation of concerns by allowing different teams or applications to manage their resources independently, reducing the risk of configuration errors and unauthorized changes.
+
+The Dashboard component lives inside the `radius-system` namespace in the Kubernetes cluster where it is installed. UCPD also resides within the same namespace.
+
+The webapp portal is accessible to various configured users. Quoting from Backstage threat model, these users could belong to one of these trust levels:
+
+**An internal user** is an authenticated user that generally belongs to the organization of a particular Backstage deployment. These users are trusted to the extent that they are not expected to compromise the availability of Backstage, but they are not trusted to not compromise data confidentiality or integrity.
```

Comment:
Can you find a way to simplify this for readers? It's not super clear that we're quoting. With all of the test here, it's a lot to read and I think readers will want to give feedback on this.

Reply:
can add a link and summarize instead.